certmanager/generate_cert.go

```diff
@@ -83,6 +83,32 @@ const (
 // See http://www.alvestrand.no/objectid/2.5.29.17.html.
 var oidSubjectAltName = asn1.ObjectIdentifier{2, 5, 29, 17}
 
+// GenCSR generates a X.509 certificate sign request with the given options.
+func GenCsr(options CertOptions) ([]byte, []byte) {
+	// Generates a CSR
+	// The public key will be bound to the certficate generated below. The
+	// private key will be used to sign this certificate in the self-signed
+	// case, otherwise the certificate is signed by the signer private key
+	// as specified in the CertOptions.
+	priv, err := rsa.GenerateKey(rand.Reader, options.RSAKeySize)
+	if err != nil {
+		glog.Fatalf("RSA key generation failed with error %s.", err)
+	}
+	template := genCsrTemplate(options)
+	csrBytes, err := x509.CreateCertificateRequest(rand.Reader, &template, crypto.PrivateKey(priv))
+	if err != nil {
+		glog.Fatalf("Could not create certificate (err = %s).", err)
+	}
+
+	// Returns the certificate that carries the RSA public key as well as
+	// the corresponding private key.
```

Comment:
The comment says the certificate carries RSA public key as well as private key. This does not sound right to me. The CSR should contain a digital signature signed with the private key. Could you clarify?